pkg/cluster/spec/prometheus.go

```diff
@@ -40,13 +40,20 @@ type PrometheusSpec struct {
 	DataDir         string               `yaml:"data_dir,omitempty"`
 	LogDir          string               `yaml:"log_dir,omitempty"`
 	NumaNode        string               `yaml:"numa_node,omitempty" validate:"numa_node:editable"`
+	RemoteConfig    Remote               `yaml:"remote_config,omitempty" validate:"remote_config:ignore"`
 	Retention       string               `yaml:"storage_retention,omitempty" validate:"storage_retention:editable"`
 	ResourceControl meta.ResourceControl `yaml:"resource_control,omitempty" validate:"resource_control:editable"`
 	Arch            string               `yaml:"arch,omitempty"`
 	OS              string               `yaml:"os,omitempty"`
 	RuleDir         string               `yaml:"rule_dir,omitempty" validate:"rule_dir:editable"`
 }
 
+// Remote prometheus remote config
+type Remote struct {
+	RemoteWrite []map[string]interface{} `yaml:"remote_write,omitempty" validate:"remote_write:ignore"`
```

Comment:
I think it should be `map[string]interface{}` by https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write

Reply:
remote-read and remote-write is array, is use  map[string]interface{}, will get error:
```
cannot unmarshal !!seq into map[string]interface {}
```

Reply:
Sorry for my careless, you can reference to @9547's idea then.

Reply:
IMHO,  @9547’s idea also work bad when remote is empty. I don't think the current encode func work bad.

And I had sent my wechat to your gmail， Can we communicate on WeChat？

Reply:
I notice that you are in stack channel `#sig-tiup`, maybe we can talk about it there

Reply:
added ut and fixed the encoded yaml file is invalid